sound/soc/intel/boards/sof_maxim_common.c

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: GPL-2.0
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+#include <sound/pcm.h>
+#include <sound/soc.h>
+#include <sound/soc-dai.h>
+#include <sound/soc-dapm.h>
+#include <uapi/sound/asound.h>
```

Comment:
@sathya-nujella you can remove the ones that are already included in sof_maxim_common.h

Reply:
Hi @ranj063 
If I partially move the header files.. I see build errors.
So, I moved all sound include files to sof_maxim_common.h and its c file only includes it.

Reply:
I wouldn't do that. Every .c and .h file has to include its required headers itself without relying on recursive inclusion.

Reply:
I went through all lines and identified needed snd headers and included them now. Please review now. Thanks!

Reply:
Hi @lyakh ,
For my learning, I want to understand this:
1) >Every .c and .h file has to include its required headers itself without relying on recursive inclusion.
Let's say in my case, if I put all snd headers in sof_maxim_common.h and sof_maxim_common.c only includes /sof_maxim_common.h.
Build wise and functionality wise it works. For my learning, asking... what kind of issue comes because of that maintainers ask to include snd headers differently like every *.c and *.h needs to include them.. 

2) regarding forwarding defining structures:
Again for my understanding.. let's say: In sof_maxim_common.h, I include related snd headers.. then we don't need to do forward define them, right?
Forward define is only needed if those header files are not included..

Please share inputs on this.

Reply:
@sathya-nujella answering to above points:
1. Not everything that works is also a good way to do it. My first pascal program - I had to print it out on paper back then - was basically filling each line from position 0 to position 79... (and mind you, it was a long and rather complex program). We don't do that today. Including all required headers in the file is basically documentation and ease of reading. You document where various parts come from and the reader has to only do one jump to respective declarations instead of following potentially multi-level nested headers. Imagine, if you include 3 headers in your .c. Each of those headers collects other headers - say, sound, pm, and others. But those headers also rely on recursive inclusion, so, to find a definition of `printk()` you first have to find a complete path through all the recursions. With this requirement it's always just one jump.
2. Sure, if you include a header with a complete declaration of a struct, you don't need to forward-declare it.

Reply:
Thanks a lot @lyakh for explaining in detail. Understood now.